recipe/conda_build_config.yaml

```diff
@@ -1,14 +1,8 @@
 python:
-  - 3.10
+  - 3.11
 python_impl:
   - cpython
 numpy:
   - 1.16
-c_compiler:                    # [win]
-  - vs2017                     # [win]
-cxx_compiler:                  # [win]
-  - vs2017                     # [win]
-vc:                            # [win]
-  - 14.1                       # [win]
 MACOSX_SDK_VERSION:            # [osx and x86_64]
   - 11.0                       # [osx and x86_64]
```

Comment:
Is this intended?  We can built it pretty nicely with 10.14 SDK on osx-64

Reply:
Yes, it's intended. Building with 10.14 will give you a python interpreter that's broken on 11.0. For eg: https://github.com/conda-forge/python-feedstock/issues/575

Reply:
However, using 11.0 doesn't build for intel 64 macos here

Reply:
We tested Python 3.7-10 which was built with SDK v10.10 on OSX v12 and didn't get the assertion that jakirkham got in the above issue. We are going to build out with 10.4 and test, will keep you updated

Reply:
If you have any more thoughts please let me know

Reply:
> However, using 11.0 doesn't build for intel 64 macos here. 

I don't understand what you are saying here. Details please.

Reply:
If we use macOS SDK version 11.0, we can't build python 3.11 for the osx-64 target. (64-bit intel CPU Mac). However, for previous versions of Python (we're on 3.10.4 at the moment), we successfully used SDK version 11.0 to build for all targets (both ARM and Intel mac).

Reply:
What is the error when you use SDK 11.0?

Reply:
```/var/folders/sy/f16zz6x50xz3113nwtb9bvq00000gp/T/abs_2092500kee/croots/recipe/python-split_1663247373378/work/Objects/longobject.c:5693:5: error: implicit declaration of function 'static_assert' is invalid in C99 [-Werror,-Wimplicit-function-declaration]```

The only change between a successful build, and a build with this failure, is changing the SDK version from 10.14 to 11.0.

The `static_assert` call in question was added somewhere between Python v3.10.4 and v3.11.

Reply:
> `implicit declaration of function 'static_assert' is invalid in C99 [-Werror,-Wimplicit-function-declaration]`

That's a consequence of upstream python [switching](https://github.com/python/cpython/pull/91733) to C11, which makes `static_assert` a keyword (where it might might have been e.g. a macro before; xref upstream [PR](https://github.com/python/cpython/pull/91730)).

Have you verified that you're compiling with `-std=c11`?

Reply:
That topic is also discussed quite extensively in https://github.com/python/cpython/issues/91731